### PR TITLE
Merge remote-tracking branch 'upstream/dev' into feat-fast-sync

### PR DESCRIPTION
**Pull request by rafal-ch**
This PR integrates the recent changes from the `dev` (as of 6.12.2021 11:00am CEST) branch into the `feat-fast-sync` branch.

### CODE REVIEW

**Review by Fraser999 — APPROVED** [review]

File: node/CHANGELOG.md [hunk 1]
```diff
@@ -29,8 +29,30 @@ All notable changes to this project will be documented in this file.  The format
...
+* The block proposer component now retains pending deploys and transfers across a restart.
+
+
+## [1.4.0] - 2021-10-04
+
+### Added
+* Add `enable_manual_sync` boolean option to `[contract_runtime]` in the config.toml which enables manual LMDB sync.
 * Add `contract_runtime_execute_block` histogram tracking execution time of a whole block.
 * Long-running events now log their event type.
 * Individual weights for traffic throttling can now be set through the configuration value `network.estimator_weights`.
```
Comment (afck) [inline node/CHANGELOG.md:56]: Fast sync and archival sync weren't added in 1.4.0. The fast sync-related points should probably be moved up to `[Unreleased]`.

Comment (rafal-ch) [reply node/CHANGELOG.md:56]: Change log rearranged here: https://github.com/casper-network/casper-node/pull/2432/commits/b952cfd5b917b8d3e27356e04473e6371dc580ed

File: node/CHANGELOG.md [hunk 2]
```diff
@@ -12,28 +12,25 @@ All notable changes to this project will be documented in this file.  The format
...
-### Fixed
-* The block proposer component now retains pending deploys and transfers across a restart.
-
-
-## [1.4.0] - 2021-10-04
-
 ### Added
 * Add new event to the main SSE server stream accessed via `<IP:Port>/events/main` which emits hashes of expired deploys.
-* Add new event to the main SSE server stream across all endpoints `<IP:PORT>/events/*` which emits a shutdown event when the node shuts down.
+* Add new event to the main SSE server stream across all endpoints `<IP:Port>/events/*` which emits a shutdown event when the node shuts down.
```
Comment (afck) [inline node/CHANGELOG.md:17]: Ah, is that only on `feat-fast-sync` and not on `dev`?

File: node/CHANGELOG.md [hunk 3]
```diff
@@ -12,28 +12,25 @@ All notable changes to this project will be documented in this file.  The format
...
+* Introducing fast-syncing to join the network, avoiding the need to execute every block to catch up.
+* Added `archival_sync` to `[node]` config section, along with archival syncing capabilities
 
 ### Changed
 * `enable_manual_sync` configuration parameter defaults to `true`.
+* Major rewrite of the contract runtime component.
```
Comment (afck) [inline node/CHANGELOG.md:23]: Not sure if that's still true?

**Review by afck — APPROVED** [review]
Looks good to me; not sure about some of the changelog entries. Maybe it would be good to verify them and look at the diff with `dev`.

File: node/CHANGELOG.md (hunk 2 above)
Comment (rafal-ch) [reply node/CHANGELOG.md:17]: It is on `dev` as well, in the `[Unreleased]` section.
https://github.com/casper-network/casper-node/blob/dev/node/CHANGELOG.md

Comment (afck) [reply node/CHANGELOG.md:17]: Ah, makes sense, thanks!

File: node/CHANGELOG.md (hunk 3 above)
Comment (rafal-ch) [reply node/CHANGELOG.md:23]: I browsed the changes in the PR when this line appeared (https://github.com/casper-network/casper-node/pull/1669/files) and yes, some of the changes are no longer true for the `fast-sync` (in particular the `chainspec_loader.rs` and `era_supervisor.rs`). But still, some of the changes remain on `dev`. So I guess it's a 50/50 :thinking:

**Review by afck — APPROVED** [review]

**Pull request merged**